pkg/oauth/oauthclient_test.go

```diff
@@ -0,0 +1,395 @@
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		testBuilder, client := generateOAuthClientBuilder(testSettings, oauthClientName)
+
+		// Test the Delete function
+		_, err := testBuilder.Create()
+		assert.Nil(t, err)
+
+		// Assert that the object actually exists
+		_, err = Pull(client, oauthClientName)
+		assert.Nil(t, err)
+	}
+}
+
+func TestOAuthClientDelete(t *testing.T) {
+	testCases := []struct {
+		addToRuntimeObjects bool
+	}{
+		{
+			addToRuntimeObjects: true,
+		},
+		{
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, &oauthv1.OAuthClient{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: oauthClientName,
+				},
+			})
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		testBuilder, client := generateOAuthClientBuilder(testSettings, oauthClientName)
+
+		// Testing the delete function
+		err := testBuilder.Delete()
+		assert.Nil(t, err)
+
+		// Assert that the object actually does not exist
+		_, err = Pull(client, oauthClientName)
+		assert.NotNil(t, err)
+	}
+}
+
+func TestOAuthClientGet(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+		{
+			expectedError:       fmt.Errorf("error: received nil OAuthClient builder"),
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testOAuthClient := generateOAuthClient(oauthClientName)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testOAuthClient)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		oauthClientBuilder, err := Pull(testSettings, oauthClientName)
+		if testCase.expectedError == nil {
+			assert.Nil(t, err)
+		}
+
+		// Test the Get function
+		builderResult, err := oauthClientBuilder.Get()
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestOAuthClientExists(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+		},
+		{
+			expectedError:       fmt.Errorf("error: OAuthClient object oauth not found"),
+			addToRuntimeObjects: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testOAuthClient := generateOAuthClient(oauthClientName)
+
```

Comment:
I think maybe you should use generateOAuthClientBuilder here? Then you can test that  oauthClientBuilder.Exists() is equal to testCase.addToRuntimeObjects